x-pack/plugins/ml/public/data_frame/components/define_pivot/define_pivot_form.tsx

```diff
@@ -66,6 +66,99 @@ export function getDefaultPivotState(kibanaContext: KibanaContextValue): DefineP
     valid: false,
   };
 }
+export function isAggNameConflict(
+  aggName: AggName,
+  aggList: PivotAggsConfigDict,
+  groupByList: PivotGroupByConfigDict
+) {
+  if (aggList[aggName] !== undefined) {
+    toastNotifications.addDanger(
+      i18n.translate('xpack.ml.dataframe.definePivot.aggExistsErrorMessage', {
+        defaultMessage: `An aggregation configuration with the name '{aggName}' already exists.`,
+        values: { aggName },
+      })
+    );
+    return true;
+  }
+
+  if (groupByList[aggName] !== undefined) {
+    toastNotifications.addDanger(
+      i18n.translate('xpack.ml.dataframe.definePivot.groupByExistsErrorMessage', {
+        defaultMessage: `A group by configuration with the name '{aggName}' already exists.`,
+        values: { aggName },
+      })
+    );
+    return true;
+  }
+
+  let conflict = false;
+
+  // check the new aggName against existing aggs and groupbys
+  const aggNameSplit = aggName.split('.');
+  let aggNameCheck: string;
+  aggNameSplit.forEach(aggNamePart => {
+    aggNameCheck = aggNameCheck === undefined ? aggNamePart : `${aggNameCheck}.${aggNamePart}`;
+    if (aggList[aggNameCheck] !== undefined || groupByList[aggNameCheck] !== undefined) {
+      toastNotifications.addDanger(
+        i18n.translate('xpack.ml.dataframe.definePivot.nestedConflictErrorMessage', {
+          defaultMessage: `Couldn't add configuration '{aggName}' because of a nesting conflict with '{aggNameCheck}'.`,
+          values: { aggName, aggNameCheck },
+        })
+      );
+      conflict = true;
+    }
+  });
+
+  if (conflict) {
+    return true;
+  }
+
+  // check all aggs against new aggName
+  Object.keys(aggList).forEach(aggListName => {
+    const aggListNameSplit = aggListName.split('.');
+    let aggListNameCheck: string;
+    aggListNameSplit.forEach(aggListNamePart => {
+      aggListNameCheck =
+        aggListNameCheck === undefined ? aggListNamePart : `${aggListNameCheck}.${aggListNamePart}`;
+      if (aggListNameCheck === aggName) {
+        toastNotifications.addDanger(
+          i18n.translate('xpack.ml.dataframe.definePivot.nestedAggListConflictErrorMessage', {
+            defaultMessage: `Couldn't add configuration '{aggName}' because of a nesting conflict with '{aggListName}'.`,
+            values: { aggName, aggListName },
+          })
+        );
+        conflict = true;
+      }
+    });
+  });
+
+  if (conflict) {
+    return true;
+  }
+
+  // check all group-bys against new aggName
+  Object.keys(groupByList).forEach(groupByListName => {
```

Comment:
The behavior of checking for conflicts in `aggLists` and `groupByList` is pretty much the same so I agree with pulling it out into a separate function in a follow-up like discussed. 👍